### PR TITLE
[LT-4360] Add withdrawal logs to AccountsManagement service.

### PR DESCRIPTION
**Pull request by fatihdumanli**
JIRA Issue link: [LT-4360](https://lykke-snow.atlassian.net/browse/LT-4360)

This PR introduces new changes to the files `WithdrawalCommandHandler` and `UpdateBalanceCommandsHandler` to include log messages about the withdrawal process.

The cases covered

- Log error message when balance couldn't be updated
- Log error message when the balance has been updated successfully
- Log error message when account couldn't be found 
- Log error message when accountCapital.Disposable < command.Amount
- Log error message when the withdrawal is disabled for the account
- Log information message when overall withdrawal operation succeeded
- Log error message when overall withdrawal operation failed

[LT-4360]: https://lykke-snow.atlassian.net/browse/LT-4360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/MarginTrading.AccountsManagement/Workflow/UpdateBalance/UpdateBalanceCommandsHandler.cs [hunk 1]
```diff
@@ -78,6 +78,10 @@ internal class UpdateBalanceCommandsHandler
                 {
                     _logger.LogWarning(ex, "Validation error while updating balance for account {AccountId}",
                         command.AccountId);
+                    
+                    _logger.LogWarning($"The account balance could not be updated during withdrawal. Reason: Validation error." +
+                        "Details: (OperationId: {OperationId}, AccountId: {AccountId}, Amount: {Amount}, Reason: {Reason})",
+                        command.OperationId, command.AccountId, command.AmountDelta);
```
Comment (anokhin-lkk) [inline src/MarginTrading.AccountsManagement/Workflow/UpdateBalance/UpdateBalanceCommandsHandler.cs:84]: {Reason} always will be empty here and it already exists above.
And in general, did you check what we will have in the logs in such a case?
In the template, you have Amount: {Amount}, but in the parameters command.AmountDelta. Will it work as expected?

File: src/MarginTrading.AccountsManagement/Workflow/Withdrawal/WithdrawalCommandsHandler.cs [hunk 2]
```diff
@@ -74,22 +82,36 @@ private async Task Handle(WithdrawCommand command, IEventPublisher publisher)
             var accountCapital = await _accountManagementService.GetAccountCapitalAsync(account.Id, useCache: false);
             if (accountCapital.Disposable < command.Amount)
             {
+                var reasonStr = $"The account {account.Id} balance {accountCapital.Balance}{accountCapital.AssetId} " +
```
Comment (anokhin-lkk) [inline src/MarginTrading.AccountsManagement/Workflow/Withdrawal/WithdrawalCommandsHandler.cs:85]: Why do we log accountCapital.Balance if we are comparing with accountCapital.Disposable?

File: src/MarginTrading.AccountsManagement/Workflow/UpdateBalance/UpdateBalanceCommandsHandler.cs (hunk 1 above)
Comment (fatihdumanli) [reply src/MarginTrading.AccountsManagement/Workflow/UpdateBalance/UpdateBalanceCommandsHandler.cs:84]: The reason parameter is cleared, I probably missed clearing it after pasting the block.

About the AmountDelta, when I checked, I saw that the AmountDelta holds the amount with the sign, which is always (-) for withdrawal. I think we can assume that AmountDelta will serve the purpose here.

```
[2023-01-05 00:54:36Z] [WRN] [::] [null] -  The account balance could not be updated during withdrawal. Reason: Validation error. Details: (OperationId: 0cd7786c-8b76-43e9-aa1e-d9b39eece66b, AccountId: 00eadd9c-b404-4fe3-ad2e-d2170e155d49, Amount: -52.93). 
```

image

File: src/MarginTrading.AccountsManagement/Workflow/Withdrawal/WithdrawalCommandsHandler.cs (hunk 2 above)
Comment (fatihdumanli) [reply src/MarginTrading.AccountsManagement/Workflow/Withdrawal/WithdrawalCommandsHandler.cs]: As per our conversation, I replaced Balance with Disposable for both the event payload and the log message.

**Review by anokhin-lkk — DISMISSED** [review]

**Review by tarurar — APPROVED** [review]

**Pull request merged**